### ISSUE TITLE
after the pod workers delete pod syncTerminated failed  ,kubelet cann't start a new pod with the same namespacedname 

### ISSUE THREAD

**Issue by dsxing** [kubernetes/kubernetes#113091]
### What happened?

Started a cluster with kubelet v1.22.5
Added a static pod yaml to /etc/kubernetes/manifest/
Observed the pod run successfully
Then remove the static pod yaml
Observed the pod is removed
And then add the same static pod yaml only change the pod id
Observed the static pod new start 

kubelet log display：
"Error syncing pod, skipping" err="mounted volumes=[cert-dir data-dir log-dir resource-dir static-pod-dir usr-local-bin]: timed out waiting for the condition"……

kubelet log also display  Cyclicly： 
"Pod cannot start yet"
"Pod cannot start yet"
……

### What did you expect to happen?

after the old pod deleted , a new pod with the same name and namespace can be start successfully

### How can we reproduce it (as minimally and precisely as possible)?

it occurs in a private cloud environment occasionally；
 the kubelet log shows  umount pod volumes failed when deleting the old pod will occurs the  bug 

### Kubernetes version

<details>

```console
$ kubectl version
v1.22.5
```

</details>

### Cloud provider

<details>
openshift
</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by lixiaobing1** [comment]
which type of volume, not unmount?

**Comment by SergeyKanzhelev** [comment]
/triage needs-information

Can this be reproed on newer k8s version? 1.22 is almost out of support. And there were fixes in pod lifecycle recently.

**Comment by dsxing** [comment]
> /triage needs-information
> 
> Can this be reproed on newer k8s version? 1.22 is almost out of support. And there were fixes in pod lifecycle recently.

i don't test on  newer k8s version.  but  I think  it is still a problem from source code

**Comment by dsxing** [comment]
> which type of volume, not unmount?



> /triage needs-information
> 
> Can this be reproed on newer k8s version? 1.22 is almost out of support. And there were fixes in pod lifecycle recently.

I don't test on newer k8s version. but I think it is still a problem from master branch code, if unmount failed or delete cgroups failed

**Comment by smarterclayton** [comment]
This should be fixed, please reopen if you can reproduce on 1.27
